src/asm/stages/stage_1.asm

```diff
@@ -4,7 +4,8 @@
 .code16
 
 _start:
-	# clear all registers
+	# clear all registers and store the boot drive
+	mov [BOOT_DRIVE], dl # The BIOS stores the boot drive in dl
```

Comment:
Is this behavior guaranteed for every BIOS?

Reply:
Trick question, some BIOSes are stupid and load bogus values(they also load you into 7c00:0000 instead of 0000:7c000...). We can have a work around by using a different BIOS function(the one GRUB uses)